app/models/course.rb

```diff
@@ -585,6 +585,8 @@ def gcse_grade_required
   end
 
   def last_published_at
+    return enrichments.select(&:last_published_timestamp_utc).max_by(&:last_published_timestamp_utc)&.last_published_timestamp_utc if enrichments.loaded?
+
     enrichments.maximum(:last_published_timestamp_utc)
   end
```

Comment:
This looks like it can be generalised for all associations.

Have you considered looking into [ActiveRecord/Associations/CollectionProxy](https://api.rubyonrails.org/v7.2/classes/ActiveRecord/Associations/CollectionProxy.html)? A `#maximum` method there might be able to address this issue across the entire app with a single change.

Reply:
Yeah that looks tempting. but I would be careful about doing that. It's something we should discuss though, it seems like a useful thing to do.